src/jvmTest/kotlin/org/treeWare/model/operator/DifferenceTests.kt

```diff
@@ -0,0 +1,120 @@
+package org.treeWare.model.operator
+
+import org.treeWare.metaModel.addressBookMetaModel
+import org.treeWare.model.assertMatchesJson
+import org.treeWare.model.decoder.stateMachine.MultiAuxDecodingStateMachineFactory
+import org.treeWare.model.decoder.stateMachine.StringAuxStateMachine
+import org.treeWare.model.encoder.EncodePasswords
+import org.treeWare.model.encoder.MultiAuxEncoder
+import org.treeWare.model.encoder.StringAuxEncoder
+import org.treeWare.model.getMainModelFromJsonString
+import org.treeWare.model.readFile
+import kotlin.test.Test
+import kotlin.test.assertNotEquals
+
+class DifferenceTests {
+    @Test
+    fun `Difference operator must return difference of its inputs`() {
+        // Ensure inputs are all different so that the test is not trivial.
+        val jsonInput1 = readFile("model/difference/mini_test_book_1.json")
+        val jsonInput2 = readFile("model/difference/mini_test_book_2.json")
+        val expectedAddTestOutput = "model/difference/mini_test_book_difference_add_1_2.json"
+        //val expectedDeleteTestOutput = "model/difference/mini_test_book_difference_delete_1_2.json" //this is empty
+        assertNotEquals(jsonInput1, jsonInput2)
+
+        val aux1 = "aux1"
+        val aux2 = "aux2"
+        val multiAuxDecodingStateMachineFactory = MultiAuxDecodingStateMachineFactory(
+            aux1 to { StringAuxStateMachine(it) },
+            aux2 to { StringAuxStateMachine(it) }
+        )
+
+        val input1 = getMainModelFromJsonString(
+            addressBookMetaModel,
+            jsonInput1,
+            multiAuxDecodingStateMachineFactory = multiAuxDecodingStateMachineFactory
+        )
+        val input2 = getMainModelFromJsonString(
+            addressBookMetaModel,
+            jsonInput2,
+            multiAuxDecodingStateMachineFactory = multiAuxDecodingStateMachineFactory
+        )
+        val output = difference(listOf(input1, input2))
+        val addOutput = output[0]
+        val deleteOutput = output[1]
+
+        assertMatchesJson(
+            addOutput!!, expectedAddTestOutput, EncodePasswords.ALL, MultiAuxEncoder(
```

Comment:
nit: do not use `!!` in code. You can avoid it here by calling `assertNotNull(addOutput)` before the call to `assertMatchesJson()`